dali/test/python/test_operator_gaussian_blur.py

```diff
@@ -0,0 +1,221 @@
+
+from test_utils import get_dali_extra_path, check_batch, RandomlyShapedDataIterator, dali_type
+
+data_root = get_dali_extra_path()
+images_dir = os.path.join(data_root, 'db', 'single', 'jpeg')
+
+
+def to_batch(tl, batch_size):
+    return [np.array(tl[i]) for i in range(batch_size)]
+
+
+def to_cv_sigma(sigma, axes=2):
+    if sigma is None:
+        return (0,) * axes
+    elif isinstance(sigma, float):
+        return (sigma,) * axes
+    elif len(sigma) == 1:
+        return (sigma[0],) * axes
+    return tuple(reversed(sigma))
+
+
+def to_cv_win_size(window_size, axes=2, sigma=None):
+    if window_size is None:
+        # when using cv2.getGaussianKernel we need to always provide window size
+        if sigma is not None:
+            sigma = to_cv_sigma(sigma, axes)
+            return tuple([int(3 * s + 0.5) * 2 + 1 for s in sigma])
+        return (0,) * axes
+    elif isinstance(window_size, int):
+        return (window_size,) * axes
+    elif len(window_size) == 1:
+        return (window_size[0],) * axes
+    # OpenCV shape is the other way round: (width, height)
+    return tuple(reversed(window_size))
+
+
+def gaussian_cv(image, sigma, window_size):
+    sigma_x, sigma_y = to_cv_sigma(sigma)
+    window_size_cv = to_cv_win_size(window_size)
+    # compute on floats and round like a sane person (in mathematically complicit way)
+    blurred = cv2.GaussianBlur(np.float32(image), window_size_cv, sigmaX=sigma_x, sigmaY=sigma_y)
+    return np.uint8(blurred + 0.5)
+
+
+def gaussian_baseline(image, sigma, window_size, axes=2, is_sequence=False, dtype=np.uint8):
+    sigma_xyz = to_cv_sigma(sigma, axes)
+    win_xyz = to_cv_win_size(window_size, axes, sigma)
+    filters = [cv2.getGaussianKernel(win_xyz[i], sigma_xyz[i]) for i in range(axes)]
+    filters = [np.float32(f).squeeze() for f in filters]
+    filters.reverse()
+    for i in reversed(range(axes)):
+        axis = i if not is_sequence else i + 1
+        image = convolve1d(np.float32(image), filters[i], axis, mode="mirror")
+    if dtype == np.float32:
+        return image
+    else:
+        return dtype(image + 0.5)
+
+
+def check_gaussian_blur(batch_size, sigma, window_size, op_type="cpu"):
+    decoder_device = "cpu" if op_type == "cpu" else "mixed"
+    pipe = Pipeline(batch_size=batch_size, num_threads=4, device_id=0)
+    with pipe:
+        input, _ = fn.file_reader(file_root=images_dir, shard_id=0, num_shards=1)
+        decoded = fn.image_decoder(input, device=decoder_device, output_type=types.RGB)
+        blurred = fn.gaussian_blur(decoded, sigma=sigma, window_size=window_size)
+        pipe.set_outputs(blurred, decoded)
+    pipe.build()
+
+    result, input = pipe.run()
+    if op_type == "gpu":
+        result = result.as_cpu()
+        input = input.as_cpu()
+    input = to_batch(input, batch_size)
+    baseline_cv = [gaussian_cv(img, sigma, window_size) for img in input]
+    check_batch(result, baseline_cv, batch_size, max_allowed_error=1)
```

Comment:
Maybe it is better to run more than one iteration?

Reply:
Done